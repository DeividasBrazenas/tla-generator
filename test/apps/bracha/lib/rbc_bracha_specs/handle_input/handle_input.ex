defmodule Generated.HandleInput do
  def handle_input(rbc, input) do
    broadcaster = rbc.broadcaster
    me = rbc.me
    output = rbc.output
    peers = rbc.peers
    propose_sent = rbc.propose_sent

    ^broadcaster = me

    ^propose_sent = false

    rbc = %{rbc | propose_sent: true}

    msgs = Enum.into(peers, %{}, fn peer_id -> {peer_id, [{:PROPOSE, me, input}]} end)

    {:ok, rbc, msgs, output}
  end
end