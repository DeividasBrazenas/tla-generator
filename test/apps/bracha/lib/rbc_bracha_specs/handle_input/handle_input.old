-------------------- MODULE handle_input --------------------
EXTENDS Naturals, FiniteSets, Sequences, TLC

CONSTANT CN
CONSTANT FN
CONSTANT Value
CONSTANT NotValue
CONSTANT NULL

(*--algorithm handle_input
variables
  bcNode \in AN,
  rbcs = [node_id \in AN |->
       [n |-> N,
       f |-> F,
       me |-> node_id,
       peers |-> AN,
       broadcaster |-> bcNode,
       predicate |-> [v \in AllValues |-> TRUE],
       max_msg_size |-> 1000,
       propose_sent |-> FALSE,
       msg_recv |-> [msg_from_node_id \in AN |-> [msg_type \in MsgTypes |-> FALSE]],
       echo_sent |-> FALSE,
       echo_recv |-> [echo_value \in AllValues |-> [x \in AN |-> FALSE]],
       ready_sent |-> FALSE,
       ready_recv |-> [ready_value \in AllValues |-> [x \in AN |-> FALSE]],
       output |-> NULL]],
  input \in (CASE bcNode \in CN -> Value [] bcNode \in FN -> {NotValue}),
  msgs = [node_id \in AN |-> {}],

define
  AN  == CN \cup FN
  N == Cardinality(AN)
  F == Cardinality(FN)
  AllValues == Value \cup {NotValue}
  MsgTypes == {"PROPOSE", "ECHO", "READY"}
end define;

fair process handle_input \in AN
variables
  rbc = rbcs[self],
  broadcaster = NULL,
  me = NULL,
  output = NULL,
  peers = NULL,
  propose_sent = NULL,
  result = NULL,
begin
handle_input:
  broadcaster := rbc.broadcaster;
  me := rbc.me;
  output := rbc.output;
  peers := rbc.peers;
  propose_sent := rbc.propose_sent;

  if broadcaster /= me then
    goto Done;
  end if;
  after_pin_0:

  if propose_sent /= FALSE then
    goto Done;
  end if;
  after_pin_1:

  rbc.propose_sent := TRUE;

  msgs := [peer_id \in peers |-> msgs[peer_id] \cup {<<"PROPOSE", me, input>>}];

  result := <<"ok", rbc, msgs, output>>;

end process;
end algorithm; *)

=============================================================