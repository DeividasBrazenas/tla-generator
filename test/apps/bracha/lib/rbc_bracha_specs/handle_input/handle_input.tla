-------------------- MODULE handle_input --------------------
EXTENDS Naturals, FiniteSets, Sequences, TLC

CONSTANT CN
CONSTANT FN
CONSTANT Value
CONSTANT NotValue
CONSTANT NULL

(*--algorithm handle_input
variables
  bcNode \in AN,
  rbcs = [node_id \in AN |->
       [n |-> N,
       f |-> F,
       me |-> node_id,
       peers |-> AN,
       broadcaster |-> bcNode,
       predicate |-> [v \in AllValues |-> TRUE],
       max_msg_size |-> 1000,
       propose_sent |-> FALSE,
       msg_recv |-> [msg_from_node_id \in AN |-> [msg_type \in MsgTypes |-> FALSE]],
       echo_sent |-> FALSE,
       echo_recv |-> [echo_value \in AllValues |-> [x \in AN |-> FALSE]],
       ready_sent |-> FALSE,
       ready_recv |-> [ready_value \in AllValues |-> [x \in AN |-> FALSE]],
       output |-> NULL]],
  input \in (CASE bcNode \in CN -> Value [] bcNode \in FN -> {NotValue}),
  msgs = [node_id \in AN |-> {}],

define
  AN  == CN \cup FN
  N == Cardinality(AN)
  F == Cardinality(FN)
  AllValues == Value \cup {NotValue}
  MsgTypes == {"PROPOSE", "ECHO", "READY"}
end define;

fair process handle_input \in AN
variables
  rbc = rbcs[self],
  broadcaster = NULL,
  me = NULL,
  output = NULL,
  peers = NULL,
  propose_sent = NULL,
  result = NULL,
begin
handle_input:
  broadcaster := rbc.broadcaster;
  me := rbc.me;
  output := rbc.output;
  peers := rbc.peers;
  propose_sent := rbc.propose_sent;

  if broadcaster /= me then
    goto Done;
  end if;
  after_pin_0:

  if propose_sent /= FALSE then
    goto Done;
  end if;
  after_pin_1:

  rbc.propose_sent := TRUE;

  msgs := [peer_id \in peers |-> msgs[peer_id] \cup {<<"PROPOSE", me, input>>}];

  result := <<"ok", rbc, msgs, output>>;

end process;
end algorithm; *)
\* BEGIN TRANSLATION (chksum(pcal) = "930ef61f" /\ chksum(tla) = "c4356351")
\* Label handle_input of process handle_input at line 50 col 3 changed to handle_input_
VARIABLES bcNode, rbcs, input, msgs, pc

(* define statement *)
AN  == CN \cup FN
N == Cardinality(AN)
F == Cardinality(FN)
AllValues == Value \cup {NotValue}
MsgTypes == {"PROPOSE", "ECHO", "READY"}

VARIABLES rbc, broadcaster, me, output, peers, propose_sent, result

vars == << bcNode, rbcs, input, msgs, pc, rbc, broadcaster, me, output, peers, 
           propose_sent, result >>

ProcSet == (AN)

Init == (* Global variables *)
        /\ bcNode \in AN
        /\ rbcs =   [node_id \in AN |->
                  [n |-> N,
                  f |-> F,
                  me |-> node_id,
                  peers |-> AN,
                  broadcaster |-> bcNode,
                  predicate |-> [v \in AllValues |-> TRUE],
                  max_msg_size |-> 1000,
                  propose_sent |-> FALSE,
                  msg_recv |-> [msg_from_node_id \in AN |-> [msg_type \in MsgTypes |-> FALSE]],
                  echo_sent |-> FALSE,
                  echo_recv |-> [echo_value \in AllValues |-> [x \in AN |-> FALSE]],
                  ready_sent |-> FALSE,
                  ready_recv |-> [ready_value \in AllValues |-> [x \in AN |-> FALSE]],
                  output |-> NULL]]
        /\ input \in (CASE bcNode \in CN -> Value [] bcNode \in FN -> {NotValue})
        /\ msgs = [node_id \in AN |-> {}]
        (* Process handle_input *)
        /\ rbc = [self \in AN |-> rbcs[self]]
        /\ broadcaster = [self \in AN |-> NULL]
        /\ me = [self \in AN |-> NULL]
        /\ output = [self \in AN |-> NULL]
        /\ peers = [self \in AN |-> NULL]
        /\ propose_sent = [self \in AN |-> NULL]
        /\ result = [self \in AN |-> NULL]
        /\ pc = [self \in ProcSet |-> "handle_input_"]

handle_input_(self) == /\ pc[self] = "handle_input_"
                       /\ broadcaster' = [broadcaster EXCEPT ![self] = rbc[self].broadcaster]
                       /\ me' = [me EXCEPT ![self] = rbc[self].me]
                       /\ output' = [output EXCEPT ![self] = rbc[self].output]
                       /\ peers' = [peers EXCEPT ![self] = rbc[self].peers]
                       /\ propose_sent' = [propose_sent EXCEPT ![self] = rbc[self].propose_sent]
                       /\ IF broadcaster'[self] /= me'[self]
                             THEN /\ pc' = [pc EXCEPT ![self] = "Done"]
                             ELSE /\ pc' = [pc EXCEPT ![self] = "after_pin_0"]
                       /\ UNCHANGED << bcNode, rbcs, input, msgs, rbc, result >>

after_pin_0(self) == /\ pc[self] = "after_pin_0"
                     /\ IF propose_sent[self] /= FALSE
                           THEN /\ pc' = [pc EXCEPT ![self] = "Done"]
                           ELSE /\ pc' = [pc EXCEPT ![self] = "after_pin_1"]
                     /\ UNCHANGED << bcNode, rbcs, input, msgs, rbc, 
                                     broadcaster, me, output, peers, 
                                     propose_sent, result >>

after_pin_1(self) == /\ pc[self] = "after_pin_1"
                     /\ rbc' = [rbc EXCEPT ![self].propose_sent = TRUE]
                     /\ msgs' = [peer_id \in peers[self] |-> msgs[peer_id] \cup {<<"PROPOSE", me[self], input>>}]
                     /\ result' = [result EXCEPT ![self] = <<"ok", rbc'[self], msgs', output[self]>>]
                     /\ pc' = [pc EXCEPT ![self] = "Done"]
                     /\ UNCHANGED << bcNode, rbcs, input, broadcaster, me, 
                                     output, peers, propose_sent >>

handle_input(self) == handle_input_(self) \/ after_pin_0(self)
                         \/ after_pin_1(self)

(* Allow infinite stuttering to prevent deadlock on termination. *)
Terminating == /\ \A self \in ProcSet: pc[self] = "Done"
               /\ UNCHANGED vars

Next == (\E self \in AN: handle_input(self))
           \/ Terminating

Spec == /\ Init /\ [][Next]_vars
        /\ \A self \in AN : WF_vars(handle_input(self))

Termination == <>(\A self \in ProcSet: pc[self] = "Done")

\* END TRANSLATION 

=============================================================
