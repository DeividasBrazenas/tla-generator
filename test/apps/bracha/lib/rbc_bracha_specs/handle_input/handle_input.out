@!@!@STARTMSG 2262:0 @!@!@
TLC2 Version 2.18 of Day Month 20?? (rev: 6dfec66)
@!@!@ENDMSG 2262 @!@!@
@!@!@STARTMSG 2187:0 @!@!@
Running breadth-first search Model-Checking with fp 46 and seed 3926952365495517933 with 1 worker on 8 cores with 7220MB heap and 64MB offheap memory [pid: 7316] (Windows 11 10.0 amd64, Oracle Corporation 18.0.1.1 x86_64, MSBDiskFPSet, DiskStateQueue).
@!@!@ENDMSG 2187 @!@!@
@!@!@STARTMSG 2220:0 @!@!@
Starting SANY...
@!@!@ENDMSG 2220 @!@!@
Parsing file C:\University\Master's Thesis\tla_generator_v2\test\apps\bracha\lib\rbc_bracha_specs\handle_input\handle_input.tla (file:/c:/University/Master's%20Thesis/tla_generator_v2/test/apps/bracha/lib/rbc_bracha_specs/handle_input/handle_input.tla)
Warning: symbols were renamed.
Parsing file C:\Users\Deividas\AppData\Local\Temp\tlc-17702165458486652758\Naturals.tla (jar:file:/C:/Users/Deividas/.vscode/extensions/alygin.vscode-tlaplus-nightly-2022.12.3006/tools/tla2tools.jar!/tla2sany/StandardModules/Naturals.tla)
Parsing file C:\Users\Deividas\AppData\Local\Temp\tlc-17702165458486652758\FiniteSets.tla (jar:file:/C:/Users/Deividas/.vscode/extensions/alygin.vscode-tlaplus-nightly-2022.12.3006/tools/tla2tools.jar!/tla2sany/StandardModules/FiniteSets.tla)
Parsing file C:\Users\Deividas\AppData\Local\Temp\tlc-17702165458486652758\Sequences.tla (jar:file:/C:/Users/Deividas/.vscode/extensions/alygin.vscode-tlaplus-nightly-2022.12.3006/tools/tla2tools.jar!/tla2sany/StandardModules/Sequences.tla)
Parsing file C:\Users\Deividas\AppData\Local\Temp\tlc-17702165458486652758\TLC.tla (jar:file:/C:/Users/Deividas/.vscode/extensions/alygin.vscode-tlaplus-nightly-2022.12.3006/tools/tla2tools.jar!/tla2sany/StandardModules/TLC.tla)
Semantic processing of module Naturals
Semantic processing of module Sequences
Semantic processing of module FiniteSets
Semantic processing of module TLC
Semantic processing of module handle_input
@!@!@STARTMSG 2219:0 @!@!@
SANY finished.
@!@!@ENDMSG 2219 @!@!@
@!@!@STARTMSG 2185:0 @!@!@
Starting... (2022-12-31 00:04:41)
@!@!@ENDMSG 2185 @!@!@
@!@!@STARTMSG 2189:0 @!@!@
Computing initial states...
@!@!@ENDMSG 2189 @!@!@
@!@!@STARTMSG 2269:0 @!@!@
Computed 2 initial states...
@!@!@ENDMSG 2269 @!@!@
@!@!@STARTMSG 2269:0 @!@!@
Computed 4 initial states...
@!@!@ENDMSG 2269 @!@!@
@!@!@STARTMSG 2190:0 @!@!@
Finished computing initial states: 7 distinct states generated at 2022-12-31 00:04:41.
@!@!@ENDMSG 2190 @!@!@
@!@!@STARTMSG 2193:0 @!@!@
Model checking completed. No error has been found.
  Estimates of the probability that TLC did not check all reachable states
  because two distinct states had the same fingerprint:
  calculated (optimistic):  val = 3.6E-15
@!@!@ENDMSG 2193 @!@!@
@!@!@STARTMSG 2201:0 @!@!@
The coverage statistics at 2022-12-31 00:04:41
@!@!@ENDMSG 2201 @!@!@
@!@!@STARTMSG 2773:0 @!@!@
<Init line 92, col 1 to line 92, col 4 of module handle_input>: 7:7
@!@!@ENDMSG 2773 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 93, col 12 to line 93, col 24 of module handle_input: 1
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2775:0 @!@!@
  line 94, col 21 to line 108, col 35 of module handle_input: 4:16
@!@!@ENDMSG 2775 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 94, col 34 to line 94, col 35 of module handle_input: 4
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 109, col 23 to line 109, col 80 of module handle_input: 4
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 109, col 28 to line 109, col 40 of module handle_input: 4
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 109, col 45 to line 109, col 49 of module handle_input: 3
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 109, col 54 to line 109, col 66 of module handle_input: 1
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2775:0 @!@!@
  |line 109, col 71 to line 109, col 80 of module handle_input: 1:2
@!@!@ENDMSG 2775 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  ||line 109, col 72 to line 109, col 79 of module handle_input: 1
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2775:0 @!@!@
  line 110, col 19 to line 110, col 41 of module handle_input: 7:28
@!@!@ENDMSG 2775 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 110, col 32 to line 110, col 33 of module handle_input: 7
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2775:0 @!@!@
  line 112, col 18 to line 112, col 45 of module handle_input: 7:28
@!@!@ENDMSG 2775 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 112, col 28 to line 112, col 29 of module handle_input: 7
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2775:0 @!@!@
  line 113, col 26 to line 113, col 47 of module handle_input: 7:28
@!@!@ENDMSG 2775 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 113, col 36 to line 113, col 37 of module handle_input: 7
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2775:0 @!@!@
  line 114, col 17 to line 114, col 38 of module handle_input: 7:28
@!@!@ENDMSG 2775 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 114, col 27 to line 114, col 28 of module handle_input: 7
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2775:0 @!@!@
  line 115, col 21 to line 115, col 42 of module handle_input: 7:28
@!@!@ENDMSG 2775 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 115, col 31 to line 115, col 32 of module handle_input: 7
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2775:0 @!@!@
  line 116, col 20 to line 116, col 41 of module handle_input: 7:28
@!@!@ENDMSG 2775 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 116, col 30 to line 116, col 31 of module handle_input: 7
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2775:0 @!@!@
  line 117, col 27 to line 117, col 48 of module handle_input: 7:28
@!@!@ENDMSG 2775 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 117, col 37 to line 117, col 38 of module handle_input: 7
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2775:0 @!@!@
  line 118, col 21 to line 118, col 42 of module handle_input: 7:28
@!@!@ENDMSG 2775 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 118, col 31 to line 118, col 32 of module handle_input: 7
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 119, col 12 to line 119, col 54 of module handle_input: 7
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2775:0 @!@!@
  |line 119, col 17 to line 119, col 54 of module handle_input: 7:28
@!@!@ENDMSG 2775 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  ||line 119, col 27 to line 119, col 33 of module handle_input: 7
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2772:0 @!@!@
<handle_input_ line 121, col 1 to line 121, col 19 of module handle_input>: 173:392
@!@!@ENDMSG 2772 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 121, col 27 to line 121, col 52 of module handle_input: 1288
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 121, col 27 to line 121, col 34 of module handle_input: 896
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 122, col 27 to line 122, col 93 of module handle_input: 392
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 123, col 27 to line 123, col 66 of module handle_input: 392
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 124, col 27 to line 124, col 78 of module handle_input: 392
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 125, col 27 to line 125, col 75 of module handle_input: 392
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 126, col 27 to line 126, col 96 of module handle_input: 392
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 127, col 30 to line 127, col 60 of module handle_input: 392
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 128, col 35 to line 128, col 71 of module handle_input: 336
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 129, col 35 to line 129, col 78 of module handle_input: 56
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 130, col 27 to line 130, col 80 of module handle_input: 392
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2772:0 @!@!@
<after_pin_0 line 132, col 1 to line 132, col 17 of module handle_input>: 22:56
@!@!@ENDMSG 2772 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 132, col 25 to line 132, col 48 of module handle_input: 952
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 132, col 25 to line 132, col 32 of module handle_input: 896
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 133, col 28 to line 133, col 54 of module handle_input: 56
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 134, col 36 to line 134, col 69 of module handle_input: 0
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 135, col 33 to line 135, col 76 of module handle_input: 56
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 136, col 25 to line 138, col 60 of module handle_input: 56
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2772:0 @!@!@
<after_pin_1 line 140, col 1 to line 140, col 17 of module handle_input>: 22:56
@!@!@ENDMSG 2772 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 140, col 25 to line 140, col 48 of module handle_input: 952
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 140, col 25 to line 140, col 32 of module handle_input: 896
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 141, col 25 to line 141, col 71 of module handle_input: 56
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 142, col 25 to line 142, col 113 of module handle_input: 56
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2775:0 @!@!@
  |line 142, col 33 to line 142, col 113 of module handle_input: 56:224
@!@!@ENDMSG 2775 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  ||line 142, col 46 to line 142, col 56 of module handle_input: 56
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 143, col 25 to line 143, col 101 of module handle_input: 56
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 144, col 25 to line 144, col 58 of module handle_input: 56
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 145, col 25 to line 146, col 67 of module handle_input: 56
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2772:0 @!@!@
<Terminating line 152, col 1 to line 152, col 11 of module handle_input>: 0:7
@!@!@ENDMSG 2772 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 152, col 40 to line 152, col 56 of module handle_input: 410
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  |line 152, col 40 to line 152, col 47 of module handle_input: 382
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 152, col 31 to line 152, col 37 of module handle_input: 224
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2221:0 @!@!@
  line 153, col 19 to line 153, col 32 of module handle_input: 7
@!@!@ENDMSG 2221 @!@!@
@!@!@STARTMSG 2202:0 @!@!@
End of statistics.
@!@!@ENDMSG 2202 @!@!@
@!@!@STARTMSG 2200:0 @!@!@
Progress(7) at 2022-12-31 00:04:41: 518 states generated (28,331 s/min), 224 distinct states found (12,251 ds/min), 0 states left on queue.
@!@!@ENDMSG 2200 @!@!@
@!@!@STARTMSG 2199:0 @!@!@
518 states generated, 224 distinct states found, 0 states left on queue.
@!@!@ENDMSG 2199 @!@!@
@!@!@STARTMSG 2194:0 @!@!@
The depth of the complete state graph search is 7.
@!@!@ENDMSG 2194 @!@!@
@!@!@STARTMSG 2268:0 @!@!@
The average outdegree of the complete state graph is 1 (minimum is 0, the maximum 4 and the 95th percentile is 3).
@!@!@ENDMSG 2268 @!@!@
@!@!@STARTMSG 2186:0 @!@!@
Finished in 1110ms at (2022-12-31 00:04:42)
@!@!@ENDMSG 2186 @!@!@
